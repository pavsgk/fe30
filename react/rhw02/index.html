<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="UTF-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  <title>Document</title>
</head>
<body>

<script>
  const sendReq = async () => {
    const result = await fetch('https://script.google.com/macros/s/AKfycbzEYFV5JBtD5d2D2swAs6i9n0SeqMxAgZRD-tJG6F3ONbSgF38kL4qeuJ-U-FMDebvxew/exec?___testS1hop1').then(res => res.json());
    console.log(result);
    return result;
  }

  sendReq();



</script>  
</body>
</html>