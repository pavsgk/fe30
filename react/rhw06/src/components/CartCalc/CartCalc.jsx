import { shallowEqual, useSelector } from "react-redux";

function CartCalc () {
  const [cart, goods] = useSelector(store => [store.cart.items, store.goods.items], shallowEqual);
  
  const objGoods = goods.map(e => e.id).reduce((acc, current, index) => {
    acc[current] = goods[index];
    return acc;
  }, {});

  const price = cart.map(e => {
    if (objGoods.hasOwnProperty(e.id)) {
      return parseFloat(objGoods[e.id].price) * e.count
    }
    return 0;})
    .reduce((acc, cur) => acc + cur, 0)
    .toFixed(2);

  return (
    <>
      <hr />
      <h3>TOTAL: {price}</h3>
    </>
  )
}

export default CartCalc;