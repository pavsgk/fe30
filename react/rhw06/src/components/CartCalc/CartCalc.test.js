import CartCalc from "./CartCalc"
import { render } from "@testing-library/react";

describe('Snapshot test', () => {
  test('Should component render', () => {
    const { asFragment } = render(<CartCalc/>);
    
    expect(asFragment()).toMatchInlineSnapshot();
  });
});