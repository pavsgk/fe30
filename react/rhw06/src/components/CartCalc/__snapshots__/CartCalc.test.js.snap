// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test Should component render 1`] = `
<DocumentFragment>
  <hr />
  <h3>
    TOTAL: 0.00
  </h3>
</DocumentFragment>
`;
